### DISCUSSION TITLE
iOS Code Signing fails

### DISCUSSION THREAD

**Discussion by gnomezgrave** [codemagic-ci-cd/codemagic-docs#1820] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Have automatic code signing setup according to the documentation.
2. Setup build triggers as "tag"
4. Create a tag in the repo and it works fine.
5. Add a new target to the XCode project (ref: [ImageNotification](https://firebase.flutter.dev/docs/messaging/apple-integration/#advanced-optional-allowing-notification-images))
6. Create a new release/tag in the repo
7. Wait...
8. And 😢 

### Expected results

The workflow should be completed without errors as before.
Previously, it worked fine with a single target (Runner). But, now after adding a new target (and also having a proper bundle identifier for that target) fails the build.

### Actual results

Workflow fails during the build phase with the below error.

```
== Building for iOS ==

> xcode-project use-profiles
Configure code signing settings
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.mobileprovision
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.provisionprofile
List available code signing certificates in keychain /Users/builder/Library/codemagic-cli-tools/keychains/27-06-23_nsyc5jev.keychain-db
Searching for files matching /Users/builder/clone/ios/**/*.xcodeproj
Failed to set code signing settings for Runner.xcodeproj


Build failed :|
Failed to set up code signing settings on Xcode projects
```

### Build id (optional)

649ad50dbf21aceceaa9bd4b

**Comment by himesh-cm** [comment]
Hello @gnomezgrave, it appears that the latest build was successful and you were able to resolve it independently. If you require further assistance, please connect with us with your build ID.

**Reply by gnomezgrave** [reply]
Yes, I reverted the changes for my XCode project using Git history to the last time it worked.

**Comment by gnomezgrave** [comment, marked as answer]
So, in case someone else faced a similar issue, here are the things that I reverted.

I reverted to the last commit that worked, took a backup of the `ios` folder, reverted back to the latest code (which doesn't work).
Then I got the diff of the project and the workspace to detect the change.

`ios/Runner.xcodeproj/project.pbxproj`

This is the XCode project version.
```
- LastUpgradeCheck = 1430;
+ LastUpgradeCheck = 1300;
```

```
- compatibilityVersion = "Xcode 15.0";
+ compatibilityVersion = "Xcode 9.3";
```

I don't know if the above changes are needed. But these changes are!

For each release, debug, and profile config in the project:

```
     CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = $inherited;
     CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE;
+    CODE_SIGN_IDENTITY = "Apple Development"; // This line
     CODE_SIGN_STYLE = Automatic;
     CURRENT_PROJECT_VERSION = 1;
```

You can open the project in a text editor and add these lines. Somehow, setting the signing on the UI didn't add these into the project (correctly).

I hope this helps someone, someday 😇

**Comment by himesh-cm** [comment]
While I’m closing our thread for now, if you have any other questions or suggestions, feel free to reply back or start a new conversation anytime. Have a nice day!
